crates/oxc_linter/src/rules/react_hooks/exhaustive_deps.rs

```diff
@@ -0,0 +1,3454 @@
+        "React Hook {hook_name} does nothing when called with only one argument."
+    ))
+    .with_label(span)
+    .with_help("Did you forget to pass an array of dependencies?")
+}
+
+fn unknown_dependencies_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+      "React Hook {hook_name} received a function whose dependencies are unknown. Pass an inline function instead."
+    ))
+    .with_label(span)
+}
+
+fn async_effect_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Effect callbacks are synchronous to prevent race conditions.")
+        .with_label(span)
+}
+
+fn missing_dependency_diagnostic(hook_name: &str, deps: &[String], span: Span) -> OxcDiagnostic {
+    let deps_pretty = if deps.len() == 1 {
+        format!("'{}'", deps[0])
+    } else {
+        let mut iter = deps.iter();
+        let all_but_last = iter
+            .by_ref()
+            .take(deps.len() - 1)
+            .map(|s| format!("'{s}'",))
+            .collect::<Vec<_>>()
+            .join(", ");
+        let last = iter.next().unwrap();
+        format!("{all_but_last}, and '{last}'")
+    };
+
+    OxcDiagnostic::warn(if deps.len() == 1 {
+        format!("React Hook {hook_name} has a missing dependency: {deps_pretty}")
+    } else {
+        format!("React Hook {hook_name} has missing dependencies: {deps_pretty}")
+    })
+    .with_label(span)
+    .with_help("Either include it or remove the dependency array.")
+}
+
+fn unnecessary_dependency_diagnostic(hook_name: &str, dep_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("React Hook {hook_name} has unnecessary dependency: {dep_name}"))
+        .with_label(span)
+        .with_help("Either include it or remove the dependency array.")
+}
+
+fn dependency_array_not_array_literal_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+        "React Hook {hook_name} was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies."
+    ))
+    .with_label(span)
+    .with_help("Use an array literal as the second argument.")
+}
+
+fn literal_in_dependency_array_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("The literal is not a valid dependency because it never changes.")
+        .with_label(span)
+}
+
+fn complex_expression_in_dependency_array_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+        "React Hook {hook_name} has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked.",
+    ))
+    .with_label(span)
+}
+
+fn dependency_changes_on_every_render_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+        "React Hook {hook_name} has a dependency array that changes every render."
+    ))
+    .with_label(span)
```

Comment:
We should suggest memoizing I think, bonus points if we can point to the original definition of the variable with a span
```suggestion
    OxcDiagnostic::warn(format!(
        "React Hook {hook_name} has a dependency array that changes every render."
    ))
    .with_help("Try memoizing this variable with `useRef` or `useCallback`.")
    .with_label(span)
```